packages/web-components/fast-components/src/styles/density.ts

```diff
@@ -0,0 +1,249 @@
+import { composedParent, DesignToken } from "@microsoft/fast-foundation";
+import { designUnit, strokeWidth } from "../design-tokens";
+
+const { create } = DesignToken;
+
+/**
+ * A Design Token representing the visual height of a single line of `base`-sized text.
+ *
+ * @remarks
+ * A variant on the type ramp for the common usage of a single line of text as in most components, adjusted for visual size rather than leading between multiple lines.
+ *
+ * For text-only and icon-only buttons to be the same height, this value should be the same as the icon size.
+ *
+ * @defaultValue 16px
+ *
+ * @public
+ */
+export const typeRampBaseVisualHeight = create<string>(
+    "type-ramp-base-visual-height"
+).withDefault("16px");
+
+// Component horizontal tokens
+
+/**
+ * The design unit multiplier for component density horizontal outer padding.
+ *
+ * @remarks
+ * Ex: `3` to set the base padding to 3 units.
+ * May be fractional.
+ *
+ * @defaultValue 3
+ *
+ * @public
+ */
+export const densityComponentHorizontalOuterUnits = create<number>(
+    "density-component-horizontal-outer-units"
+).withDefault(3);
+
+/**
+ * The design unit multiplier for component density horizontal padding between elements.
+ *
+ * @remarks
+ * Ex: `2` to set the base padding to 2 units.
+ * May be fractional.
+ *
+ * @defaultValue 2
+ *
+ * @public
+ */
+export const densityComponentHorizontalBetweenUnits = create<number>(
+    "density-component-horizontal-between-units"
+).withDefault(2);
+
+/**
+ * The design unit multiplier for adjustments to component horizontal density tokens, which adds the specified design units to the base setting.
+ *
+ * @remarks
+ * Ex: `1` to increase padding by one unit.
+ * May be fractional.
+ *
+ * @defaultValue 0
+ *
+ * @public
+ */
+export const densityComponentHorizontalAdjustmentUnits = create<number>(
+    "density-component-horizontal-adjustment-units"
+).withDefault(0);
+
+/**
+ * The calculated cumulative value for adjustments to component horizontal density relative up the document hierarchy.
+ *
+ * @remarks
+ * To use relative density do not change this value, only consume it in styles.
+ * If you want to break the chain of relative density, override this default function with a fixed value.
+ *
+ * @public
+ */
+export const densityComponentHorizontalAdjustmentUnitsCumulative = create<number>(
+    "density-component-horizontal-adjustment-units-cumulative"
+);
+densityComponentHorizontalAdjustmentUnitsCumulative.withDefault(
```

Comment:
Nick and I chatted about the idea of adding native support for more complicated design tokens like this in the underlying infrastructure. The token infrastructure is designed to get around limitations of plain css properties, and one of the limitations is no way to access parent properties. I've done that here, but it's somewhat complicated and requires a shadow token `Cumulative` rather than a behavior or mode of the single `Units` token.

On the other hand, this structure does easily allow for `Cumulative` to default to being relative to the parent, or redefining this derived token to be a fixed value. In that usage "Cumulative" is probably not the correct word, so I'm open to other suggestions that mean "this is the actual value you should use within the component", whether that implies a relative or fixed value.